integration/integration_test.go

```diff
@@ -275,32 +275,100 @@ func TestPublishMessage(t *testing.T) {
 	}
 
 	unique := "UniqueStringToLookForInOutput"
-	sampleValues, err := helpers.NewStruct(map[string]interface{}{
-		"string": unique,
-		"number": 42,
+	events, err := createEvents([]string{unique})
+	if err != nil {
+		t.Fatalf("error creating events: %s\n", err)
+	}
+
+	_, err = client.PublishEvents(env.ctx, &messages.PublishRequest{
+		Events: events,
 	})
+	if err != nil {
+		t.Fatalf("Error publishing event: %s", err)
+	}
+
+	found, err = env.WaitUntil("stdout", unique)
+	if err != nil {
+		t.Fatalf("Error looking for publish results: %s", err)
+	}
+
+	if !found {
+		stderr, err := env.GetStderr()
+		if err != nil {
+			t.Errorf("Error getting standard error: %s", err)
+		}
+		stdout, err := env.GetStdout()
+		if err != nil {
+			t.Errorf("Error getting standard out: %s", err)
+		}
+		t.Fatalf("Event wasn't published.\nstderr:\n%s\nstdout:\n%s\n", stderr, stdout)
+	}
+}
+
+func TestDiskQueue(t *testing.T) {
```

Comment:
Do we need to add something here to ensure the events actually made it into a disk queue? If this configuration generated a memory queue underneath the test would still pass. The same goes for the encrypted queue test below.

If we have other tests that verify that the correct queue type is actually created from a configuration, then it is fine for these tests to just trust that the right thing happened, but I'm not sure we have those.

Reply:
Added checks for starting of disk queue